newton/sim/builder.py

```diff
@@ -3389,6 +3506,46 @@ def finalize(self, device: Devicelike | None = None, requires_grad: bool = False
             m.muscle_points = wp.array(self.muscle_points, dtype=wp.vec3, requires_grad=requires_grad)
             m.muscle_activations = wp.array(self.muscle_activations, dtype=wp.float32, requires_grad=requires_grad)
 
+            # -----------------------
+            # sites, tendons, and tendon actuators
+
+            m.site_name = self.site_name
```

Comment:
Let's keep using the word "key" here, so the variable names should be `site_key`, `tendon_key`, etc. same for method arguments.